File: Corte1/RetosEntrega19Agosto/RetoMapaCalorSalaKnuth.cpp
#include <iostream>
using namespace std;

int main() {
    // Se crean las variables a usar
    int usoHora = 0;
    int usoMaxHora = 0;
    int columnaMax = 0;
    int filaMax = 0;
    int usoDia = 0;
    int UsoTotalDia = 0;

    int usoHorarioSala[5][6] = {{usoHora, usoHora, usoHora, usoHora, usoHora, usoHora},
                             {usoHora, usoHora, usoHora, usoHora, usoHora, usoHora},
                             {usoHora, usoHora, usoHora, usoHora, usoHora, usoHora},
                             {usoHora, usoHora, usoHora, usoHora, usoHora, usoHora},
                             {usoHora, usoHora, usoHora, usoHora, usoHora, usoHora}};

    // Se usa un fro anidado para pedir todos los datos
    for (int i = 0; i < 5; i++) {
        cout << "Escriba los 6 datos de la sala para el dia " << i + 1 << ": ";
        for (int j = 0; j < 6; j++) {
            cin >> usoHora;
            usoHorarioSala[i][j] = usoHora;
        }
    }
    // Se usa un for anidado para recorrer la matriz y encontrar el uso máximo de la sala dia y hora incluidos
    for (int i = 0; i < 5; i++) {
        for (int j = 0; j < 6; j++) {
            usoHora = usoHorarioSala[i][j];
            if (usoHora > usoMaxHora) {
                usoMaxHora = usoHora;
                filaMax = i;
                columnaMax = j;
            }
        }
    }
    cout << "El uso máximo de la sala fue de " << usoMaxHora << " en el dia " << filaMax + 1 << " y la hora " << columnaMax + 1 << endl;
    // FOr anidado que revisa  el uso total de usuarios por dia y dice el mayor
    for (int i = 0; i < 5; i++) {
        for (int j = 0; j < 6; j++) {
            usoDia = usoDia + usoHorarioSala[i][j];
            if (usoDia > UsoTotalDia) {
                UsoTotalDia = usoDia;
                filaMax = i;
            }
        }
        usoDia = 0; // Reiniciar el uso diario para el siguiente día
        
    }
    cout << "El dia que mas se uso la sala fue el dia " << filaMax + 1 << " con un total de " << UsoTotalDia << " usuarios." << endl;
    // For que revisa las franjas y dias con menos de 5 usuarios
     for (int i = 0; i < 5; i++) {
        for (int j = 0; j < 6; j++) {
            if (usoHorarioSala[i][j] <= 5) {
            cout << "El dia " << i + 1 << " y la hora " << j + 1 << " tuvo menos de 5 usuarios" << endl;
            }
        }
    }
    return 0;
}

File: Corte1/RetosEntrega19Agosto/RetoMapaCalorSalaKnuth.py
usoHora = 0
franjaCongestion = 0
totalDia = 0
totalMaxDia = 0
UsoHorarioSalaKnuth = [[usoHora, usoHora, usoHora, usoHora, usoHora, usoHora],
                       [usoHora, usoHora, usoHora, usoHora, usoHora, usoHora],
                       [usoHora, usoHora, usoHora, usoHora, usoHora, usoHora],
                       [usoHora, usoHora, usoHora, usoHora, usoHora, usoHora],
                       [usoHora, usoHora, usoHora, usoHora, usoHora, usoHora]]
#Se le pide al usuario que ingrese cada dato usando for anidado para cada uno
for i in range(5):
    for j in range(6):
        inputHora = int(input(f"Ingrese el uso horario de la sala Knuth para el día {i + 1} y la hora {j + 1}: "))
        UsoHorarioSalaKnuth[i][j] = inputHora

for i in range(5):
    for j in range(6):
        #Se busca la franja horaria con mayor congestion, no menciona multiples franjas con el mismo valor, solo toma la primera
        if UsoHorarioSalaKnuth[i][j] > franjaCongestion:
            franjaCongestion = UsoHorarioSalaKnuth[i][j]
            diaCongestion = i
            horaCongestion = j
#Se dice el dia yla hora de mayor congestion numericamente y cuando se uso, para que diga el dia se puede hacer un switch o varios ifs
print(f"La franja horaria con mayor congestion es el día {diaCongestion + 1} y la hora {horaCongestion + 1} con un uso horario de {franjaCongestion}.")

#Se crea un for anidado que sume el uso total de cada dia y si es mayor lo reemplaza, al salir del bucle se imprime el dia con mayor gente y cuantos fueron
for i in range(5):
    for j in range(6):
        totalDia = totalDia + UsoHorarioSalaKnuth[i][j]
    if totalDia > totalMaxDia:
        totalMaxDia = totalDia
        diaMax = i
    totalDia = 0
print(f"El día con mayor uso, es el día {diaMax + 1} con un total de {totalMaxDia} de uso de usuarios.")

for i in range(5):
    for j in range(6):
        if UsoHorarioSalaKnuth[i][j] < 5:
            print(f"El día {i + 1} y la hora {j + 1} tuvo un uso menor a 5 personas.")